sonnet/src/conv_test.py

```diff
@@ -183,20 +183,26 @@ def testUnknownBatchSizeNCHW(self):
     out2 = defun_conv(tf.ones([5, 3, 5, 5]))
     self.assertEqual(out2.shape, [5, 2, 5, 5])
 
-  @parameterized.parameters(True, False)
-  def testUnknownChannels(self, autograph):
-    x = tf.TensorSpec([3, 3, 3, None], dtype=tf.float32)
+  def test_conv_nd_raises_on_dynamic_channels(self):
+    """Tests that ConvND raises a clear error with a dynamic channel dim."""
+    conv_module = conv.Conv2D(output_channels=3, kernel_shape=3)
 
-    c = conv.ConvND(
-        num_spatial_dims=2,
-        output_channels=1,
-        kernel_shape=3,
-        data_format="NHWC")
-    defun_conv = tf.function(c, autograph=autograph)
+    # Define a function that calls the module with a dynamic input shape.
+    # The `input_signature` specifies that the channel dimension is unknown.
+    @tf.function(input_signature=[
+        tf.TensorSpec(shape=[None, 16, 16, None], dtype=tf.float32)
+    ])
```

Comment:
This feels like the test is fairly similar to the code that was there before. Are you sure that this change is causing a behavior change?